### PR TITLE
Signaling servers sends signature with O packet - Resolves#114

### PR DESCRIPTION
**Pull request by MonsieurWave**
- before sending O packet, SS loads fileinfo from cache and randomly selects a seed-hash pair from seed2hash
- O packet now contains signature{seed, hash, size}
- on o packet fileinfo is preloaded into cache
- modified testing engine to account for varying seeds and hashes
- modified tests to account for signature in O

why? security 

resolves#114

### CODE REVIEW

File: src/FileInfoHandler/FileInfoLoader.js [hunk 1]
```diff
@@ -35,11 +37,11 @@ class FileInfoLoader {
         this.signatureGenerator.sign(stream), // return hash that has been generated
         computeFileSize(stream)
       ]);
-
       logger.log('fileinfoloader-successful', { url, seed2hash, size });
       return { seed2hash, size }; // return FileInfo
     })
     .catch((err) => {
+      console.error(err); // eslint-disable-line no-console
```
Comment (MonsieurWave) [inline src/FileInfoHandler/FileInfoLoader.js]: c'est utile de voir cette erreur
si tu aimes pas, tu peux l'enlever

Comment (GuillaumeLeclerc) [inline src/FileInfoHandler/FileInfoLoader.js]: Should be changed by debug

File: src/FileInfoHandler/UrlCache.js [hunk 2]
```diff
@@ -7,7 +7,13 @@ import dataloader from 'dataloader';
 
 import FileInfoLoader from './FileInfoLoader';
 
-
+/**
+*  UrlCache stores FileInfo (seed2hash, size) for all files
+*
+*  @param {Function} constructor - if FileInfo can not be loaded from cache,
```
Comment (GuillaumeLeclerc) [inline src/FileInfoHandler/UrlCache.js]: This is a class not a function

File: src/WebSocketConnection.js [hunk 3]
```diff
@@ -29,6 +30,9 @@ type RTCConnection = { // eslint-disable-line flowtype/type-id-match
   ices: Array<ICECandidateType>
 }
 
+// define urlCache to be used for all connections
+const urlCache = new UrlCache();
```
Comment (GuillaumeLeclerc) [inline src/WebSocketConnection.js]: Flow type would be neat

File: test/isolatedDeletionFromUrlCache.spec.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ describe('Isolated: Isolated reloading of File Info after deletion from UrlCache
     promisedFileInfo1 = await urlCache.load(testFile);
   });
   before((done) => {
-    setTimeout(done, 1000); // wait for deletion
+    setTimeout(done, 1200); // wait for deletion
```
Comment (GuillaumeLeclerc) [inline test/isolatedDeletionFromUrlCache.spec.js]: This value should be linked with `DATABRIDGE_URLCHACHE_TTL`

File: test/scenari/ConnexionFromUsertoSameUser.spec.js [hunk 5]
```diff
@@ -55,7 +55,12 @@ new Scenario({
         type: 'offer',
         sdp: 'sdlkfh'
       },
-      ices: ['asligf']
+      ices: ['asligf'],
+      signature: {
+        seed: /[a-f 0-9]{16}/,
```
Comment (GuillaumeLeclerc) [inline test/scenari/ConnexionFromUsertoSameUser.spec.js]: no space allowed

File: test/scenari/UserDecoReco.spec.js [hunk 6]
```diff
@@ -54,7 +54,12 @@ new Scenario({
         type: 'offer',
         sdp: 'ih'
       },
-      ices: ['sdfrsdf']
+      ices: ['sdfrsdf'],
+      signature: {
+        seed: /[a-f 0-9]{16}/,
```
Comment (GuillaumeLeclerc) [inline test/scenari/UserDecoReco.spec.js]: Same here

**Comment by MonsieurWave** [comment]
Requested changes implemented

**Pull request merged**